src/stream/min_by.rs

```diff
@@ -0,0 +1,54 @@
+use std::cmp::Ordering;
+use std::pin::Pin;
+
+use super::stream::Stream;
+use crate::future::Future;
+use crate::task::{Context, Poll};
+
+/// A future that yields the minimum item in a stream by a given comparison function.
+#[derive(Clone, Debug)]
+pub struct MinBy<S: Stream, F> {
+    stream: S,
+    compare: F,
+    min: Option<S::Item>,
+}
+
+impl<S: Stream + Unpin, F> Unpin for MinBy<S, F> {}
+
+impl<S: Stream + Unpin, F> MinBy<S, F> {
+    pub(super) fn new(stream: S, compare: F) -> Self {
+        MinBy {
+            stream,
+            compare,
+            min: None,
+        }
+    }
+}
+
+impl<S, F> Future for MinBy<S, F>
+where
+    S: futures_core::stream::Stream + Unpin,
```

Comment:
Ah, this bound may need to be updated to just be `Stream` as #140 was merged.

Reply:
hmmm, you probably mean #145 ?

Reply:
Ah yes indeed

Reply:
Is this still unresolved?